### ISSUE TITLE
convert_bigquery_policy_to_iam generating a "KeyError: 'role'" error

### ISSUE THREAD

**Issue by lukwam** [forseti-security/forseti-security#3099]
Since upgrading to Forseti `v2.19.0` about a week and a half ago, I'm getting lots of errors about `KeyError: 'role'` for the `convert_bigquery_policy_to_iam` process.

From Cloud Error Reporting:

![image](https://user-images.githubusercontent.com/1765418/63283073-3dcca380-c27e-11e9-8fad-7f0220a11caa.png)

Currently using the default bigquery rules defined here, with the organization_id set to our org id.

https://github.com/forseti-security/forseti-security/blob/dev/rules/bigquery_rules.yaml

/l

**Comment by kevensen** [comment]
@lukwam Thank you for reporting this issue.  I am looking into this now.

**Comment by kevensen** [comment]
@lukwam Do you mind providing the error from the Stackdriver logs?

For your reference: https://forsetisecurity.org/docs/latest/faq/index.html#where-are-the-logs-stored-for-forseti

**Comment by lukwam** [comment]
```textPayload:  "[forseti-security][2.19.0] google.cloud.forseti.services.inventory.base.iam_helpers(convert_bigquery_policy_to_iam): unknown role in access policy..." ```

Removed the user and the project name from the rest of the error.

The role listed in the policy is `'*'`.

**Comment by kevensen** [comment]
@lukwam That helps to provide some context, thank you.  I should have been more clear; can you please provide the stack trace as well?  It should be in the Stackdriver log message.

**Comment by lukwam** [comment]
```KeyError: 'role'
at convert_bigquery_policy_to_iam (/usr/local/lib/python3.6/dist-packages/forseti_security-2.19.0-py3.6.egg/google/cloud/forseti/services/inventory/base/iam_helpers.py:70)
at get_dataset_policy (/usr/local/lib/python3.6/dist-packages/forseti_security-2.19.0-py3.6.egg/google/cloud/forseti/services/inventory/base/resources.py:1031)
at wrapper (/usr/local/lib/python3.6/dist-packages/forseti_security-2.19.0-py3.6.egg/google/cloud/forseti/services/inventory/base/resources.py:109)
at visit (/usr/local/lib/python3.6/dist-packages/forseti_security-2.19.0-py3.6.egg/google/cloud/forseti/services/inventory/crawler.py:120)
at accept (/usr/local/lib/python3.6/dist-packages/forseti_security-2.19.0-py3.6.egg/google/cloud/forseti/services/inventory/base/resources.py:336)
at try_accept (/usr/local/lib/python3.6/dist-packages/forseti_security-2.19.0-py3.6.egg/google/cloud/forseti/services/inventory/base/resources.py:302)```

**Comment by kevensen** [comment]
Thanks for the stack trace.  On your server VM, can you please:

1. Sudo to root

`sudo -i`

2. Change directory to the forseti-security source directory

`cd /home/ubuntu/forseti-security`

3. Open `iam_helpers.py` for edit

`vi /home/ubuntu/forseti-security/google/cloud/forseti/services/inventory/base/iam_helpers.py`

4. Edit lines 69 and 70

https://github.com/forseti-security/forseti-security/blob/cbebc4a0479dcb10930df19c993737997815506b/google/cloud/forseti/services/inventory/base/iam_helpers.py#L69-L70

To resemble the following
```python
    for policy in access_policy:
        LOGGER.info('POLICY: %s', str(policy))
        if policy['role'] not in access_policy_to_iam_role_map:
```
5. Exit your editor

6. Restart Forseti

`systemctl restart forseti`

7. Create a new inventory, model, run a new scan, and notify.

8. Please provide the new log output.

**Comment by lukwam** [comment]
The new thing is this:

```"[forseti-security][2.19.0] google.cloud.forseti.services.inventory.base.resources(get_iam_policy): Could not get Dataset IAM Policy: Bigquery Dataset IAM policy is not supported by this API client"```

**Comment by kevensen** [comment]
Thanks for the update.  Just a few questions.

1. Did this occur before or after you added the new log line?
2. Did that new log line produce new output?
3. How did you initially install Forseti?  The Python installer or Terraform?

**Comment by lukwam** [comment]
1. I don't see any occurrences of that `Could not get Dataset IAM Policy: Bigquery Dataset IAM policy is not supported by this API client` from before I made the change you described. 
2. When I search for `text:"POLICY:"` those are the only entries I see.
3. This is a long-standing forseti instance that I've had since early 2.0 days and I've steadily upgraded from version to version and it pre-exists the terraform setup which I haven't moved to yet.

**Comment by kevensen** [comment]
@lukwam I truly apologize for not getting back to you sooner.  Are you still experiencing this error?  If so, can you please provide me the entire Stackdriver entry for:

`"[forseti-security][2.19.0] google.cloud.forseti.services.inventory.base.resources(get_iam_policy): Could not get Dataset IAM Policy: Bigquery Dataset IAM policy is not supported by this API client"`

**Comment by gkowalski-google** [comment]
Hi @lukwam, I am able to reproduce this issue. Can you confirm if you have created any [authorized views](https://cloud.google.com/bigquery/docs/share-access-views) in this project? If you are not sure, you can:

1. Go to the [BigQuery console](https://console.cloud.google.com/bigquery)
2. For each dataset:
  2a. Select the dataset
  2b. Click Share Dataset (in the middle/right of the screen)
  2c. Click the Authorized Views tab

Assuming that you can confirm this is the case, then this might be an easy fix and I will try to have it included in the next release.

**Comment by gkowalski-google** [comment]
@lukwam, I did some more testing and it looks like this issue will only happen if you have the Cloud Asset Inventory (CAI) disabled. Since you have been using Forseti a while and upgrading it, my guess is that this was never enabled.

I recommend that you enable the CAI inventory setting, as we are continuously adding more Cloud resources into the inventory to be scanned. I also recommend giving Terraform a shot, as it will simplify the upgrade process going forward and we are planning on not supporting the Python installer at the end of this month. Using Terraform will make it easier to enable this CAI setting as well.

Let me know if you'd like to setup some time, and I can help with the Terraform installation.

**Comment by gkowalski-google** [comment]
This fix has been included in Forseti v2.21.0.
